--- a/etc/init.d/network
+++ b/etc/init.d/network
@@ -5,6 +5,104 @@ STOP=90
 
 USE_PROCD=1
 
+fix_wireless_config() {
+        if [ -d '/sys/class/pci_bus/0000:01/device/0000:01:00.0' ];then
+                source '/sys/class/pci_bus/0000:01/device/0000:01:00.0/uevent'
+		touch /etc/config/wireless
+		local need_fix=0
+                if [ "$PCI_ID" == "14C3:7906" ];then
+			if ! grep '3c0000000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0' /etc/config/wireless >/dev/null;then
+				need_fix=1
+			fi
+			if ! grep '3c0000000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0+1' /etc/config/wireless >/dev/null;then
+				need_fix=1
+			fi
+			if grep "option htmode 'HE80'" /etc/config/wireless >/dev/null;then
+				need_fix=1
+			fi
+			if grep "option encryption 'none'" /etc/config/wireless >/dev/null;then
+				need_fix=1
+			fi
+                        if [ $need_fix -eq 1 ];then
+                                cat > /etc/config/wireless <<EOF
+
+config wifi-device 'radio0'
+        option type 'mac80211'
+        option path '3c0000000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0'
+        option country 'US'
+        option legacy_rates '1'
+        option mu_beamformer '0'
+        option band '2g'
+        option htmode 'HT40'
+        option channel '11'
+        option noscan '1'
+
+config wifi-iface 'default_radio0'
+        option device 'radio0'
+        option network 'lan'
+        option mode 'ap'
+        option ssid 'H69K_Wireless_2.4G'
+        option encryption 'psk2'
+        option key 'password'
+
+config wifi-device 'radio1'
+        option type 'mac80211'
+        option path '3c0000000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0+1'
+        option channel '36'
+        option band '5g'
+        option country 'US'
+        option legacy_rates '1'
+        option mu_beamformer '0'
+        option htmode 'HE80'
+
+config wifi-iface 'default_radio1'
+        option device 'radio1'
+        option network 'lan'
+        option mode 'ap'
+        option key 'password'
+        option ssid 'H69K_Wireless_5G'
+        option encryption 'psk2'
+
+EOF
+                        fi
+                else
+			if ! grep '3c0000000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0' /etc/config/wireless >/dev/null;then
+				need_fix=1
+			fi
+                       if grep '3c0000000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0+1' /etc/config/wireless > /dev/null;then
+				need_fix=1
+			fi
+			if grep "option encryption 'none'" /etc/config/wireless >/dev/null;then
+				need_fix=1
+			fi
+			if [ $need_fix -eq 1 ];then
+                                cat > /etc/config/wireless <<EOF
+
+config wifi-device 'radio0'
+        option type 'mac80211'
+        option path '3c0000000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0'
+        option band '5g'
+        option country 'US'
+        option legacy_rates '1'
+        option mu_beamformer '0'
+        option channel '36'
+        option htmode 'VHT80'
+
+config wifi-iface 'default_radio0'
+        option device 'radio0'
+        option network 'lan'
+        option mode 'ap'
+        option ssid 'H69K_Wireless'
+        option encryption 'psk2'
+        option key 'password'
+
+EOF
+                        fi
+                fi
+		sync
+        fi
+}
+
 init_switch() {
 	setup_switch() { return 0; }
 
@@ -13,6 +111,7 @@ init_switch() {
 }
 
 start_service() {
+	fix_wireless_config
 	init_switch
 
 	procd_open_instance
