--- ./etc/init.d/boot.orig	2020-02-14 16:51:11.914391316 +0800
+++ ./etc/init.d/boot	2020-02-14 16:56:02.761499250 +0800
@@ -22,6 +22,9 @@ boot() {
 	[ -f /proc/jffs2_bbc ] && echo "S" > /proc/jffs2_bbc
 	[ -f /proc/net/vlan/config ] && vconfig set_name_type DEV_PLUS_VID_NO_PAD
 
+	ulimit -n 131072
+	echo 1000000 > /sys/devices/system/cpu/cpufreq/policy0/scaling_min_freq 
+
 	mkdir -p /var/run
 	mkdir -p /var/log
 	mkdir -p /var/lock
