--- a/etc/init.d/sysfixtime	2023-04-08 21:57:58.707491705 +0800
+++ b/etc/init.d/sysfixtime	2023-04-08 21:57:06.599492680 +0800
@@ -6,6 +6,7 @@ STOP=90
 
 RTC_DEV=/dev/rtc0
 HWCLOCK=/sbin/hwclock
+FIX_CPU_FREQ=/usr/sbin/fixcpufreq.pl
 
 boot() {
 	hwclock_load
@@ -18,6 +19,7 @@ boot() {
 }
 
 start() {
+	[ -x "$FIX_CPU_FREQ" ] && "$FIX_CPU_FREQ"
 	hwclock_load
 }
 
@@ -26,7 +28,11 @@ stop() {
 }
 
 hwclock_load() {
-	[ -e "$RTC_DEV" ] && [ -e "$HWCLOCK" ] && $HWCLOCK -s -u -f $RTC_DEV
+	if ${HWCLOCK} | grep '1970' > /dev/null;then
+	    return
+	else
+	    [ -e "$RTC_DEV" ] && [ -e "$HWCLOCK" ] && $HWCLOCK -s -u -f $RTC_DEV
+	fi
 }
 
 hwclock_save(){
